src/main/scala/com/picnicml/doddlemodel/preprocessing/StandardScaler.scala

```diff
@@ -0,0 +1,31 @@
+package com.picnicml.doddlemodel.preprocessing
+
+import breeze.linalg.*
+import breeze.stats.{mean, stddev}
+import com.picnicml.doddlemodel.base.Transformer
+import com.picnicml.doddlemodel.data.{Dataset, Features, RealVector, Target}
+
+/** An immutable preprocessor that transforms features by subtracting the mean and scaling to unit variance.
+  *
+  * Examples:
+  * val preprocessor = StandardScaler()
+  */
+@SerialVersionUID(1L)
+class StandardScaler private (val meanStdDev: Option[MeanStdDev])
+  extends Transformer[StandardScaler] with Serializable {
+
+  override def isFitted: Boolean = this.meanStdDev.isDefined
+
+  override def fit(x: Features, y: Target): StandardScaler =
+    new StandardScaler(Some(MeanStdDev(mean(x(::, *)).t, stddev(x(::, *)).t)))
+
+  override protected def transformSafe(x: Features, y: Target): Dataset =
+    ((x(*, ::) - meanStdDev.get.mean).apply(*, ::) / meanStdDev.get.stdDev, y)
```

Comment:
We should probably check if stdDev is 0 (the check in scikit: https://github.com/scikit-learn/scikit-learn/blob/a24c8b46/sklearn/preprocessing/data.py#L61)

Reply:
Nice catch! Should be fixed now.